js/packages/mobile-wallet-adapter-protocol/src/transact.ts

```diff
@@ -171,7 +173,18 @@ export async function transact<TReturn>(
                         state.associationPublicKey,
                         state.ecdhPrivateKey,
                     );
-                    state = { __type: 'connected', sharedSecret };
+                    const sessionPropertiesBuffer = responseBuffer.slice(ENCODED_PUBLIC_KEY_LENGTH_BYTES);
```

Comment:
Testing my understanding here:

In v1 protocol (MWA 2.0), the wallet will include a `session_properties` in their `responseBuffer`. 

It will now be `responseBuffer = public_key + session_properties` (all encoded).